### PR TITLE
fix: forward timeout waitForFileChooser

### PR DESCRIPTION
**Pull request by OrKoN**

### CODE REVIEW

File: src/util/DeferredPromise.ts [hunk 1]
```diff
@@ -31,10 +31,13 @@ export function createDeferredPromiseWithTimer<T>(
     resolver = resolve;
     rejector = reject;
   });
-  const timeoutId = setTimeout(() => {
-    isRejected = true;
-    rejector(new TimeoutError(timeoutMessage));
-  }, timeout);
+  const timeoutId =
+    timeout > 0
```
Comment (jrandolf) [inline src/util/DeferredPromise.ts:35]: Could we do this check earlier and just return a normal deferred promise?

Comment (OrKoN) [reply src/util/DeferredPromise.ts:35]: How much earlier do you propose to check for it?

Comment (jrandolf) [reply src/util/DeferredPromise.ts:35]: I've made the changes.

File: src/common/NetworkManager.ts [hunk 2]
```diff
@@ -144,9 +144,9 @@ export class NetworkManager extends EventEmitter {
     if (this.#deferredInitPromise) {
       return this.#deferredInitPromise;
     }
-    this.#deferredInitPromise = createDebuggableDeferredPromise<void>(
-      'NetworkManager initialization timed out'
-    );
+    this.#deferredInitPromise = createDeferredPromise<void>({
```
Comment (OrKoN) [inline src/common/NetworkManager.ts]: this is not a debuggable promise anymore?

File: src/util/DeferredPromise.ts [hunk 3]
```diff
@@ -11,18 +11,27 @@ export interface DeferredPromise<T> extends Promise<T> {
...
  */
-export function createDeferredPromiseWithTimer<T>(
-  timeoutMessage: string,
-  timeout = 5000
-): DeferredPromise<T> {
+export function createDeferredPromise<T>({
+  message = '',
+  timeout = 5000,
+}: DeferredPromiseOptions = {}): DeferredPromise<T> {
+  if (DEFERRED_PROMISE_DEBUG_TIMEOUT > 0 && !timeout) {
```
Comment (OrKoN) [inline src/util/DeferredPromise.ts]: hm, I don't think we want to debug all deferred promises

**Review by jrandolf-google — APPROVED** [review]

**Pull request merged**